File: yologram-api-v2/app/domain/search/tech/publisher/message/tech_indexing_message.py
from dataclasses import dataclass

# 인덱싱 대상 구분 — worker가 이 값으로 분기한다. 큐를 대상별로 나누지 않고 이 필드로 가른다
TARGET_TECH_POST = "TECH_POST"


@dataclass(frozen=True)
class TechIndexingMessage:
    """
    게시글 인덱싱 작업 메시지 (api-v1 TechIndexingMessage·worker 구독 계약과 문자열로 미러).

    단건도 from == to로 보내 범위 인덱싱과 같은 경로를 탄다.
    필드명(target·from·to)이 세 프로젝트의 계약이라 한쪽만 바꾸면 소비가 깨진다.
    """

    from_id: int
    to_id: int
    target: str = TARGET_TECH_POST

    def to_payload(self) -> dict:
        # from은 파이썬 예약어라 필드명을 from_id로 두고 직렬화 시점에 계약 이름으로 바꾼다
        return {"target": self.target, "from": self.from_id, "to": self.to_id}

File: yologram-api-v1/src/main/kotlin/link/yologram/api/v1/domain/search/tech/publisher/message/TechIndexingMessage.kt
package link.yologram.api.v1.domain.search.tech.publisher.message

/**
 * 인덱싱 작업 메시지 (SQS 페이로드) — worker와 문자열 계약으로 미러.
 *
 * 예: {"target":"TECH_POST","from":1,"to":20}
 *
 * 단건도 from == to로 보낸다 — 범위 인덱싱 한 경로로 처리해 코드가 갈리지 않게 한다.
 * 레거시(yologram-legacy)는 단건을 SQS 없이 동기 처리해 경로가 둘로 나뉘었고,
 * 그 결과 단건 경로에만 문서 변환 누락 버그가 남아 있었다(문서로 변환해두고 엔티티를 인덱싱).
 *
 * target으로 대상을 구분한다 — 게시글·뉴스가 한 큐를 공유하고, INVEST_POST가 늘어도 큐를 새로 만들지 않는다.
 * 큐를 나누지 않는 이유: 인덱싱은 어드민이 가끔 누르는 작업이라 대상별로 격리할 만큼 트래픽이 없고,
 * 워커 구독자도 한 곳에서 분기하는 편이 대상 추가 비용이 낮다.
 */
data class TechIndexingMessage(
    val target: String = TARGET_TECH_POST,
    val from: Long,
    val to: Long,
) {
    companion object {
        const val TARGET_TECH_POST = "TECH_POST"
        const val TARGET_TECH_NEWS = "TECH_NEWS"
    }
}
